lib/passport-configurator.js

```diff
@@ -300,15 +300,29 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
                 }
               } else {
                 user.hasPassword(password, function(err, ok) {
-                  if (ok) {
-                    done(null, userProfile);
+                  if (ok) {          
+         if (!user.emailVerified) 
+         {
+              // Fail to log in if email verification is not done yet
+              return done(null, false, {message: 'User Email has not been verified.'});
+              err = new Error('login failed as the email has not been verified');
+              err.statusCode = 401;
+              err.code = 'LOGIN_FAILED_EMAIL_NOT_VERIFIED';
+              fn(err);
+          } 
+          else 
+          {
+                   user.createAccessToken(accessToken);
```

Comment:
I thought we wanted to create an accessToken each time the user is logged, so the local account will be like the third party accounts, where the accessToken, userId, ttl and time are stored in the db each time the user is logged in.